### PR TITLE
Predict changes

### PR DESCRIPTION
**Pull request by timcdlucas**
Make some changes to predictions.

Make the default be to predict from the data that is saved in the fitted model object (this is a standard).
Add a newdata argument that lets you predict over an arbitrary set of covariate rasters.
Add the original data to the fitted model object. In particular this keeps the mesh with the model (the model is useful without the mesh if there is a field).

Few other small things.
Add tests.

### CODE REVIEW

File: R/prepare_data.R [hunk 1]
```diff
@@ -53,10 +53,15 @@ prepare_data <- function(polygon_shapefile,
...
+
+  # Save raster layer names so we can reassign it to make sure names don't change.
+  cov_names <- names(covariate_rasters)
+
   # If no aggregation raster is given, use a 'unity' raster
   if(is.null(aggregation_raster)) {
     aggregation_raster <- covariate_rasters[[1]]
-    raster::setValues(aggregation_raster, rep(1, raster::ncell(aggregation_raster)))
+    aggregation_raster <- raster::setValues(aggregation_raster, rep(1, raster::ncell(aggregation_raster)))
+    names(aggregation_raster) <- 'aggregation_raster'
```
Comment (aknandi) [inline R/prepare_data.R]: Should setting the name of the aggregation raster be outside the if statement. So it has the same name whether it was user defined or not?

File: R/plotting.R [hunk 2]
```diff
@@ -33,6 +33,7 @@ plot.disag.data <- function(x, ...) {
 
 plot.fit.result <- function(x, ...){
   
+  parameter <- sd <- obs <- pred <- NULL
```
Comment (aknandi) [inline R/plotting.R]: What does this line do?

Comment (timcdlucas) [reply R/plotting.R]: This stops the warning in R CMD check (or `check()`). Those objects are used in non standard evaluation in ggplot or dplyr calls. Like `ggplot(d, aes(x, y))`. `x` and `y` aren't actually objects but the R check tests whether all used objects exist.

 I think there's a more official way to do it but I can't remember.

File: R/prepare_data.R (hunk 1 above)
Comment (timcdlucas) [reply R/prepare_data.R]: Yes probably.

File: R/predict.R [hunk 3]
```diff
@@ -1,39 +1,58 @@
...
+  if(model_output$model_setup$field){
+    # Extract field values
+    field <- (Amatrix %*% pars$nodemean)[, 1]
+    field_ras <- raster::rasterFromXYZ(cbind(coords, field))
+    linear_pred <- linear_pred + field_ras
+  } else {
+    field_ras <- NULL
+  }
   
   mean_prediction <- 1 / (1 + exp(-1 * linear_pred))
```
[the remark this answers is not in the document]
Comment (timcdlucas) [reply R/predict.R]: Was there supposed to be a suggestion for this bit?
```
linear_pred <- linear_pred + field_ras
  } else {
    field_ras <- NULL
  }
```

**Pull request merged**